### PR TITLE
De-crazy the expression editor function helper popover

### LINKED ISSUE #28668: [Epic] De-crazy the expression editor function helper popover
We want to improve expression editor helper popover to make it less overwhelming and more clean and helpful.
To achieve that we want to change it's design and structure.

<table>
<tr>
<th> Before: </th>
<th> After: </th>
</tr>
<tr>
<td>

Screenshot 2023-02-27 at 19 34 09

</td>
<td>

Screenshot 2023-02-27 at 19 41 14

</td>
</tr>
</table>

**Links**
- [product doc](https://www.notion.so/metabase/De-crazy-the-expression-editor-function-helper-popover-c43d8ff0b0ca4271b9c36ec2be844fe0)
- [design](https://www.figma.com/file/68g8S9L4cMesbctGfCySl7/De-crazy-the-expression-editor-function-helper-popover?node-id=0%3A1&t=1d5CQFMXJ6PhbMu8-0)
- feature branch: `28668-expression-editor`

**Implementation Plan**

- [x] [Refactored ExpressionWidget, ClauseStep components to TS](https://github.com/metabase/metabase/pull/28687)
- [x] [Change expression editor title, Implement info tooltip](https://github.com/metabase/metabase/pull/28776) 
- [x] [Change popover structure to match proposed design](https://github.com/metabase/metabase/pull/28920)
- [x] [Add interactivity for popover parameters](https://github.com/metabase/metabase/pull/28966)

**How to Test This Feature Out**

1. Start creating a new Question: click "New" button in the top bar -> Question.
2. Click "Custom column" in the Notebook.
3. Appeared expression editor should have new design and experience.

### PR DESCRIPTION
**Pull request by deniskaber**
Closes https://github.com/metabase/metabase/issues/28668, check it for full details

### Description

We want to improve expression editor helper popover to make it less overwhelming and more clean and helpful.
To achieve that we want to change it's design and structure.

<table>
<tr>
<th> Before: </th>
<th> After: </th>
</tr>
<tr>
<td>

Screenshot 2023-02-27 at 19 34 09

</td>
<td>

Screenshot 2023-02-27 at 19 41 14

</td>
</tr>
</table>

### How to verify

1. Start creating a new Question: click "New" button in the top bar -> Question.
2. Click "Custom column" in the Notebook.
3. Appeared expression editor should have new design and experience.

### Demo

https://user-images.githubusercontent.com/7983744/223769602-b2deaf75-3e47-4891-92c6-f76644abf793.mov

### Checklist

- [x] [Refactored ExpressionWidget, ClauseStep components to TS](https://github.com/metabase/metabase/pull/28687)
- [x] [Change expression editor title, Implement info tooltip](https://github.com/metabase/metabase/pull/28776) 
- [x] [Change popover structure to match proposed design](https://github.com/metabase/metabase/pull/28920)
- [x] [Add interactivity for popover parameters](https://github.com/metabase/metabase/pull/28966)

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28678)

### CODE REVIEW

**Review by mazameli — CHANGES_REQUESTED** [review]
On custom filter expressions, the helper popover seems to be tethered to the bottom of the parent popover. The expectation is that it would instead appear right below or above the input box. This seems to be happening when the browser viewport is large enough to show the helper popover below.

image

Same thing with the custom summarization expression. The expectation is this would appear right below the text input for the expression. 
image

Here's a full-screen image of the size of my browser window (weird flex, I know):
image

**Comment by mazameli** [comment]
I did not spec this, but a thought that occurred to me is that we could try this icon so that it turns blue on hover, and would be clickable and would take you to [the documentation](https://www.metabase.com/docs/latest/questions/query-builder/expressions) on click. We could add to the tooltip the sentence, `Click for documentation.`

Could we try that?

image

**Comment by ranquild** [comment]
@deniskaber I think we should address this
```
I did not spec this, but a thought that occurred to me is that we could try this icon so that it turns blue on hover, and would be clickable and would take you to [the documentation](https://www.metabase.com/docs/latest/questions/query-builder/expressions) on click. We could add to the tooltip the sentence, Click for documentation.

Could we try that?
```
in the scope of this epic. This is a regression caused by this integration branch - we effectively remove the help link for expression.

We need to check whether this can be reproduced in master. If not, then we need to fix this here, too:
```
Another bug I found: updating an existing custom column will incorrectly duplicate the column. It should replace it instead. See this recording: https://www.loom.com/share/7b57e79f763e4b95ad0f28576d627c18
```

There is no rush to get this epic completed.

**Comment by mazameli** [comment]
@deniskaber will do

File: frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+          <Tooltip
+            tooltip={t`You can reference columns here in functions or equations, like: floor([Price] - [Discount]). Click for documentation.`}
+            placement="right"
+            maxWidth={332}
+          >
+            <InfoLink
+              target="_blank"
+              href={EXPRESSIONS_DOCUMENTATION_URL}
+              data-testid="expression-widget-info-link"
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx:83]: I believe this is from https://github.com/metabase/metabase/pull/29113. I've checked and seems nobody's using this anymore, so it should be removed.

**Review by mazameli — APPROVED** [review]
The description tooltip and link look good to me, and my understanding is that we'll fix that preexisting issue with the popover tethering in a different PR.

**Review by ranquild — APPROVED** [review]

**Pull request merged**